controllers/link.go

```diff
@@ -0,0 +1,59 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/nimble-link/backend/models"
+	"github.com/nimble-link/backend/pkg/linkutils"
+	"github.com/nimble-link/backend/services/authentication"
+)
+
+type ShortLinkInput struct {
+	OriginalURL string `json:"original_url" binding:"required"`
+	Alias       string `json:"alias"`
+	Password    string `json:"password"`
+}
+
+func CreateLink(c *gin.Context) {
+	var input ShortLinkInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
```

Comment:
for failing to JSON binding, I think we should use `http. StatusUnprocessableEntity` instead. What do you think?

Reply:
Good point, just know about `StatusUnprocessableEntity` :D, thank you